src/cljs/harja/ui/modal.cljs

```diff
@@ -39,7 +39,8 @@
 
 (defn- modal-container* [optiot sisalto]
   (let [{:keys [otsikko otsikon-alle-komp otsikko-tyyli footer nakyvissa? luokka
-                leveys sulje-fn sulje-ruksista-fn content-tyyli body-tyyli modal-luokka]} optiot
+                leveys sulje-fn sulje-ruksista-fn content-tyyli body-tyyli modal-luokka
+                esta-rastista-tabilla-pois-siirtyminen? siirra-fokus-rastista]} optiot
         sulje!  #(do
```

Comment:
Itse preferoisin tapaa missä ei tarvitse miettiä id:tä erikseen elementeille eli hyödyntäisi vaikka
 active-elementtiä. tässä artikkeli aiheesta :[blogautus]( https://www.bennadel.com/blog/4097-restoring-activeelement-focus-after-a-user-interaction-in-javascript.htm)

Ei tarvitsisi kaikkia modaaleja Harjassa käydä erikseen läpi ja etuna olisi myös että olisi aina toimiva kun uutta modaalia lisätään. 

Testasin että fokuksen asettaminen takaisin näyttäisi toimivan näinkin pienellä muutoksella: 
```clojure
 edellinen-elementti (.-activeElement js/document) ;; poimitaan aktiivinen elementti
     _ (prn (.-outerHTML edellinen-elementti)) ;; printti että näkee mikä fokuksessa
     sulje!  #(do
                  ;; estää file-open dialogin poistamisen
                  #_(.preventDefault %)
                  (.stopPropagation %)
                  (when sulje-fn
                    (sulje-fn))
                  (piilota!)
                  (.focus edellinen-elementti))  ;; Tässä fokus edelliseen
``` 
                  
Fokuksen asettaminen itse modaaliin pitäisi sitten onnistua :ref kautta mutta ennen kun sinne asti menee niin tuleeko mieleen jotain estettä mieleen tälläiselle lähestymiselle ylipäätään kun olet enemmän pyöritellyt tätä?                

Tämä pistäisi uusiksi pullarin logiikkaa mutta vähentäisi työtä jatkossa joten sillä nostin yhä esille tämän vaihtoehdon.

Reply:
Tämä voisi kyllä useimmissa kohdissa hyvinkin toimia, voisin vielä kokeilla paremmin. Tuota :ref myös jo tutkin ja siinä pitäisi napin koodissa lisätä :ref toiminnon funktioon siinä vaiheessa kun nappi sen tietää, ja sitä en vielä tehnyt, mutta voisin sitäkin yrittää vielä.